### ISSUE TITLE
cannot be determined query result  by using 'IS NULL' 

### ISSUE THREAD

**Issue by jiangzhx** [apache/arrow-datafusion#6279]
### Describe the bug

- `select count(1) from t1 where (select t1.c1 from t1 where c2<0) IS NULL`
- `select count(1) from t1 where (select t1.c1 from t1 where c2>0) IS NULL`

response some result.

### To Reproduce

```
CREATE EXTERNAL TABLE t1 (
    c1  VARCHAR NOT NULL,
    c2  INT NOT NULL,
    c3  SMALLINT NOT NULL,
    c4  SMALLINT NOT NULL,
    c5  INT NOT NULL,
    c6  BIGINT NOT NULL,
    c7  SMALLINT NOT NULL,
    c8  INT NOT NULL,
    c9  BIGINT NOT NULL,
    c10 VARCHAR NOT NULL,
    c11 FLOAT NOT NULL,
    c12 DOUBLE NOT NULL,
    c13 VARCHAR NOT NULL
)
STORED AS CSV
WITH HEADER ROW
LOCATION 'testing/data/csv/aggregate_test_100.csv';
```

- `select count(1) from t1 where (select t1.c1 from t1 where c2<0) IS NULL`
- `select count(1) from t1 where (select t1.c1 from t1 where c2>0) IS NULL`

### Expected behavior

shoud not response some result

### Additional context

datafusion 19.0.0

**Comment by alamb** [comment]
What is the expected output? That the query runs and produces results?

**Comment by Weijun-H** [comment]
```
DataFusion CLI v24.0.0
❯ CREATE EXTERNAL TABLE t1 (
    c1  VARCHAR NOT NULL,
    c2  INT NOT NULL,
    c3  SMALLINT NOT NULL,
    c4  SMALLINT NOT NULL,
    c5  INT NOT NULL,
    c6  BIGINT NOT NULL,
    c7  SMALLINT NOT NULL,
    c8  INT NOT NULL,
    c9  BIGINT NOT NULL,
    c10 VARCHAR NOT NULL,
    c11 FLOAT NOT NULL,
    c12 DOUBLE NOT NULL,
    c13 VARCHAR NOT NULL
)
STORED AS CSV
WITH HEADER ROW
LOCATION 'testing/data/csv/aggregate_test_100.csv';
0 rows in set. Query took 0.002 seconds.
❯ select count(1) from t1 where (select t1.c1 from t1 where c2<0) IS NULL;
This feature is not implemented: Physical plan does not support logical expression (<subquery>)
```
I tried to reproduce this error, but I am not sure what it shou be.

**Comment by jiangzhx** [comment]
> 

t's a mistake. Just happened on Datafusion 19.0.0
i will close this pr.

**Comment by jiangzhx** [comment]
> ```
> DataFusion CLI v24.0.0
> ❯ CREATE EXTERNAL TABLE t1 (
>     c1  VARCHAR NOT NULL,
>     c2  INT NOT NULL,
>     c3  SMALLINT NOT NULL,
>     c4  SMALLINT NOT NULL,
>     c5  INT NOT NULL,
>     c6  BIGINT NOT NULL,
>     c7  SMALLINT NOT NULL,
>     c8  INT NOT NULL,
>     c9  BIGINT NOT NULL,
>     c10 VARCHAR NOT NULL,
>     c11 FLOAT NOT NULL,
>     c12 DOUBLE NOT NULL,
>     c13 VARCHAR NOT NULL
> )
> STORED AS CSV
> WITH HEADER ROW
> LOCATION 'testing/data/csv/aggregate_test_100.csv';
> 0 rows in set. Query took 0.002 seconds.
> ❯ select count(1) from t1 where (select t1.c1 from t1 where c2<0) IS NULL;
> This feature is not implemented: Physical plan does not support logical expression (<subquery>)
> ```
> 
> I tried to reproduce this error, but I am not sure what it should be.

about `Physical plan does not support logical expression (<subquery>)`
i will fix at #5907

**Comment by alamb** [comment]
Thanks @jiangzhx  -- I think @mingmwang  or @jackwener  were thinking about working on this one too
